.travis.yml

```diff
@@ -1,5 +1,239 @@
 language: python
-dist: trusty
+
+matrix:
+  include:
+  - os: linux
+    python: 3.6
+    env: PYSMT_SOLVER="all"       PYSMT_GMPY="TRUE"
+  - os: linux
+    python: 3.6
+    env: PYSMT_SOLVER="all"       PYSMT_GMPY="FALSE"
+  - os: linux
+    python: 3.6
+    env: PYSMT_SOLVER="None"      PYSMT_GMPY="FALSE"   PYSMT_CYTHON="TRUE"
+  - os: linux
+    python: 3.6
+    env: PYSMT_SOLVER="None"      PYSMT_GMPY="FALSE"   PYSMT_CYTHON="FALSE"
+  - os: linux
+    python: 3.6
+    env: PYSMT_SOLVER="msat"      PYSMT_GMPY="FALSE"
+  - os: linux
+    python: 3.6
+    env: PYSMT_SOLVER="z3"        PYSMT_GMPY="FALSE"
+  - os: linux
+    python: 3.6
+    env: PYSMT_SOLVER="cvc4"      PYSMT_GMPY="FALSE"
+  - os: linux
+    python: 3.6
+    env: PYSMT_SOLVER="yices"     PYSMT_GMPY="FALSE"
+  - os: linux
+    python: 3.6
+    env: PYSMT_SOLVER="bdd"       PYSMT_GMPY="FALSE"
+  - os: linux
+    python: 3.6
+    env: PYSMT_SOLVER="picosat"   PYSMT_GMPY="FALSE"
+  - os: linux
+    python: 3.6
+    env: PYSMT_SOLVER="btor"      PYSMT_GMPY="FALSE"
+  - os: linux
+    python: 3.6
+    env: PYSMT_SOLVER="msat_wrap" PYSMT_GMPY="FALSE"
+  - os: linux
+    python: 3.6
+    env: PYSMT_SOLVER="z3_wrap"   PYSMT_GMPY="FALSE"
+
+  - os: linux
+    python: 2.7
+    env: PYSMT_SOLVER="all"       PYSMT_GMPY="TRUE"
+  - os: linux
+    python: 2.7
+    env: PYSMT_SOLVER="all"       PYSMT_GMPY="FALSE"
+  - os: linux
+    python: 2.7
+    env: PYSMT_SOLVER="None"      PYSMT_GMPY="FALSE"   PYSMT_CYTHON="TRUE"
+  - os: linux
+    python: 2.7
+    env: PYSMT_SOLVER="None"      PYSMT_GMPY="FALSE"   PYSMT_CYTHON="FALSE"
+  - os: linux
+    python: 2.7
+    env: PYSMT_SOLVER="msat"      PYSMT_GMPY="FALSE"
+  - os: linux
+    python: 2.7
+    env: PYSMT_SOLVER="z3"        PYSMT_GMPY="FALSE"
+  - os: linux
+    python: 2.7
+    env: PYSMT_SOLVER="cvc4"      PYSMT_GMPY="FALSE"
+  - os: linux
+    python: 2.7
+    env: PYSMT_SOLVER="yices"     PYSMT_GMPY="FALSE"
+  - os: linux
+    python: 2.7
+    env: PYSMT_SOLVER="bdd"       PYSMT_GMPY="FALSE"
+  - os: linux
+    python: 2.7
+    env: PYSMT_SOLVER="picosat"   PYSMT_GMPY="FALSE"
+  - os: linux
+    python: 2.7
+    env: PYSMT_SOLVER="btor"      PYSMT_GMPY="FALSE"
+  - os: linux
+    python: 2.7
+    env: PYSMT_SOLVER="msat_wrap" PYSMT_GMPY="FALSE"
+  - os: linux
+    python: 2.7
+    env: PYSMT_SOLVER="z3_wrap"   PYSMT_GMPY="FALSE"
+
+  - os: linux
+    python: pypy
+    env: PYSMT_SOLVER="None"      PYSMT_GMPY="FALSE"   PYSMT_CYTHON="TRUE"
+  - os: linux
+    python: pypy
+    env: PYSMT_SOLVER="None"      PYSMT_GMPY="FALSE"   PYSMT_CYTHON="FALSE"
+
+
+  # - os: osx
+  #   osx_image: xcode9.3
+  #   language: generic
+  #   python: 3.6
+  #   env: PYSMT_SOLVER="all"       PYSMT_GMPY="TRUE"   TRAVIS_PYTHON_VERSION="3.6.5"
```

Comment:
Ah, it would actually be nice to include testing for GMPY and CYTHON. I think PYSMT_SOLVER can take a list "msat;z3" to simulate the "all" value.

Reply:
:+1: